0001-typedb/02-dev/02-schema.md

```diff
@@ -0,0 +1,948 @@
+---
+pageTitle: Schema
+keywords: typeql, schema, type hierarchy, reserved keywords
+longTailKeywords: typeql schema, typeql type hierarchy, typeql data model, typeql reserved keyword
+Summary: Introduction to the TypeDB Schema.
+---
+
+# Schema
+
+TypeDB schema is like a blueprint of a database.
+
+TypeDB schema defines all user-defined types and rules used in a database.
+
+All user-defined types must subtype the entity, relation, or attribute base type or another user-defined type.
+
+As a result, TypeDB schemas look like ER diagrams. They resemble logical data models,
+removing the need to create a separate physical data model and allowing developers to
+create schemas that mirror their object model simplifying both design and development.
+
+![Types hierarchy](../../images/schema/thing-era-role-rule.png)
+
+TypeDB schemas are written with TypeQL which is both Data Definition Language and Data Manipulation Language (used 
+for schema and data queries). You can see a complete set of TypeQL syntax and grammar rules in ANTLR v.4 specification 
+file: [TypeQL.g4](https://github.com/vaticle/typeql/blob/master/grammar/TypeQL.g4).
+
+## Types
+
+Types define the data model, its constraints, and the relationships within it. In addition, the types within a schema 
+reflect the vocabulary used to describe the business domain it represents.
+
+Creating a type means subtyping a base type or a user-defined type. The base types are:
+
+|      Type      |              Instance of Type               |
+|:--------------:|:-------------------------------------------:|
+|  Entity type   |    Entity <br/>(Instance of Entity type)    |
+| Relation type  |  Relation <br/>(Instance of Relation type)  |
+| Attribute type | Attribute <br/>(Instance of Attribute type) |
+
+There is a strict hierarchy of types with all types being children (grandchildren, grand- grandchildren, etc.) 
+of a base type. Every type can have only one parent. There is also an internal type `role` that is used in relations. 
+Instances of types (i.e., the data) are known as `entities`, `relations`, and `attributes`.
+
+### Thing type
+
+There is a base type called `thing` type that can be used to address all types (both base and user-defined).
+
+<div class="note">
+[Warning]
+The `thing` base type will be deprecated in TypeDB version 3.0.
+</div>
+
+### Entity types
+
+Entity types define the objects within the business domain (e.g., an organization, file, or user). Further, an entity 
+(as in instance of an entity type e.g., Vaticle LTD) can exist independently of any other instance of any type 
+(e.g., John Doe).
```

Comment:
I do not understand this example

Reply:
I have rewritten this part. https://github.com/vaticle/docs/commit/f81214c3303500311df1dc1ae6be223474eb6031#diff-be80f09573928c8d94fac728eebecdaa57b7679f8dab37db8f3d82b5edc04c0cR57